cmd/kubeadm/app/cmd/join.go

```diff
@@ -618,15 +670,21 @@ func fetchInitConfigurationFromJoinConfiguration(cfg *kubeadmapi.JoinConfigurati
 }
 
 // fetchInitConfiguration reads the cluster configuration from the kubeadm-admin configMap
-func fetchInitConfiguration(tlsBootstrapCfg *clientcmdapi.Config) (*kubeadmapi.InitConfiguration, error) {
-	// creates a client to access the cluster using the bootstrap token identity
-	tlsClient, err := kubeconfigutil.ToClientSet(tlsBootstrapCfg)
-	if err != nil {
-		return nil, errors.Wrap(err, "unable to access the cluster")
+func fetchInitConfiguration(dryRunClient clientset.Interface, tlsBootstrapCfg *clientcmdapi.Config) (*kubeadmapi.InitConfiguration, error) {
+	var err error
+
+	client := dryRunClient
+	if dryRunClient == nil {
```

Comment:
Is it possible to move the creating client logic to `InitCfg()` ?

Reply:
yes, makes sense

Reply:
moved this to fetchInitConfigurationFromJoinConfiguration as that func also needs tlsBootstrapCfg.